src/main/java/io/hops/util/featurestore/FeaturestoreHelper.java

```diff
@@ -1440,29 +1440,34 @@ public static Boolean validatePrimaryKey(Dataset<Row> featuregroupDf, List<Strin
    * @param dtypes the schema of the provided spark dataframe
    * @param description the description about the feature group/training dataset
    */
-  public static void validateMetadata(String name, Tuple2<String, String>[] dtypes, String description) {
-    Pattern namePattern = Pattern.compile("^[a-zA-Z0-9_]+$");
-    if (name.length() > 256 || name.equals("") || !namePattern.matcher(name).matches())
+  public static void validateMetadata(String name, Tuple2<String, String>[] dtypes, String description,
+    FeaturestoreClientSettingsDTO featurestoreClientSettingsDTO) {
+    Pattern namePattern = Pattern.compile(featurestoreClientSettingsDTO.getFeaturestoreRegex());
+    if (name.length() > 256 || name.equals("") || !namePattern.matcher(name).matches()) {
```

Comment:
You can condense all these checks in a single one, the regex. As you are taking care of the name lengths in another PR, you can at least remove the `name.equals("")` as it's already covered by the regex.

Reply:
Removed the `equals("")` since it's being taken care of by the string. The length will be taken care of in another PR.